app/services/update_site_setting.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class UpdateSiteSetting
+  include Service::Base
+
+  policy :current_user_is_admin
+
+  step :convert_name_to_sym
+
+  policy :setting_is_visible
+  policy :setting_is_configurable
+
+  step :cleanup_value
+  step :save
+
+  private
+
+  def convert_name_to_sym(setting_name:)
+    context.setting_name = setting_name.to_sym
+  end
+
+  def current_user_is_admin(current_user:)
```

Comment:
Instead of using `current_user` here you can use the auto-injected `guardian` and do `guardian.admin?`

Reply:
Also adding `allow_changing_hidden` with default false here too would be good. Would we use `allow_changing_hidden` for stuff like hosted-site do you think?

Reply:
> Also adding `allow_changing_hidden` with default false here too would be good.

I'm not following this? When would we need to involve `allow_changing_hidden` when ensuring current user is admin?

> Would we use `allow_changing_hidden` for stuff like hosted-site do you think?

Yes, possibly. I've added it because I need it in my /about admin config area PR which I'll create shortly.